src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/request/RaptorRoutingRequestTransitDataCreator.java

```diff
@@ -43,7 +44,7 @@ class RaptorRoutingRequestTransitDataCreator {
     ZonedDateTime transitSearchTimeZero
   ) {
     this.transitLayer = transitLayer;
-    this.departureDate = transitSearchTimeZero.toLocalDate();
+    this.departureDate = ServiceDateUtils.asServiceDay(transitSearchTimeZero);
```

Comment:
This is a bit complex, since we already know the date for the trip search one frame up in the call chain. Would it make sense to pass it separately, rather than compute it again from a computed value?

Reply:
Or rather we could pass in the date here, and compute the transitSearchTimeZero from the date and transitLayer timezone